lib/opam_build.ml

```diff
@@ -76,26 +76,33 @@ let install_project_deps ~opam_files ~selection =
   let network = ["host"] in
   (if Variant.arch variant |> Ocaml_version.arch_is_32bit then
      [shell ["/usr/bin/linux32"; "/bin/sh"; "-c"]] else [])
-  @ distro_extras @ [
+  @ distro_extras
+  @ match opam_version with
+    | `V2_0 -> []
```

Comment:
To keep compatibility in case the base images upgrade to opam 2.1 by default it is better to do it anyway in any case:
```suggestion
    | `V2_0 -> [ run "sudo ln -f /usr/bin/opam-2.0 /usr/bin/opam" ]
```